File: app/models/UserIdent.java
package models;

import javax.persistence.Embeddable;

/**
 * 삭제된 유저에 대비하여
 *
 * @author Keesun Baik
 */
@Embeddable
public class UserIdent {

    public Long id;
    public String loginId;
    public String name;

    public UserIdent(User author) {
        id = author.id;
        loginId = author.loginId;
        name = author.name;
    }
}

File: src/main/kotlin/com/github/yonaprojects/yona/domain/user/UserIdent.kt
package com.github.yonaprojects.yona.domain.user

import jakarta.persistence.Embeddable

@Embeddable
class UserIdent(
    var id: Long? = null,
    var loginId: String? = null,
    var name: String? = null
) {
    constructor(user: User) : this(user.id, user.loginId, user.name)
}
